### PR TITLE
[SPARK-26117][FOLLOW-UP][SQL]throw SparkOutOfMemoryError intead of SparkException in UnsafeHashedRelation

### PR DESCRIPTION
**Pull request by heary-cao**
## What changes were proposed in this pull request?

When build hash Map with one row of data and run out of memory, we should throw a SparkOutOfMemoryError exception, which is more accurate than SparkException. this PR fix it.

## How was this patch tested?

N / A

### CODE REVIEW

**Comment by heary-cao** [comment]
cc @cloud-fan @gatorsmile @kiszk

**Comment by heary-cao** [comment]
retest this please

**Comment by kiszk** [comment]
retest this please

**Comment by kiszk** [comment]
Is this follow-up of #23084?

File: sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala [hunk 1]
```diff
@@ -24,7 +24,8 @@ import com.esotericsoftware.kryo.io.{Input, Output}
 
 import org.apache.spark.{SparkConf, SparkEnv, SparkException}
 import org.apache.spark.internal.config.MEMORY_OFFHEAP_ENABLED
-import org.apache.spark.memory.{MemoryConsumer, StaticMemoryManager, TaskMemoryManager}
+import org.apache.spark.memory.{MemoryConsumer, SparkOutOfMemoryError,
```
Comment (kiszk) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala]: Is it better to use `_`?

**Comment by kiszk** [comment]
LGTM excepts two comments

**Review by gatorsmile — COMMENTED** [review]
Please change the PR title to `[SPARK-26117] [FOLLOW-UP][SQL]throw SparkOutOfMemoryError intead of SparkException in UnsafeHashedRelation`

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Comment by heary-cao** [comment]
thanks, @cloud-fan @gatorsmile @kiszk.

**Pull request closed without merging**